src/foam/nanos/notification/Notification.js

```diff
@@ -114,5 +120,49 @@ foam.CLASS({
       name: 'slackMessage',
       documentation: 'Message to be sent to Slack if sendSlackMessage is enabled.'
     }
+  ],
+
+  methods: [
+    {
+      name: 'checkOwnership',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      type: 'Boolean',
+      javaCode: `
+        foam.nanos.auth.User user = (foam.nanos.auth.User) x.get("user");
+        if( user == null ) return false;
+
+        return getUserId() == user.getId();
+      `
+    },
+    {
+      name: 'authorizeOnCreate',
+      javaCode: `
+      AuthService auth = (AuthService) x.get("auth");
+      if ( ! auth.check(x, "notificationDAO.read.*") ) throw new AuthorizationException("You don't have permission to create this notification.");
+      `
+    },
+    {
+      name: 'authorizeOnUpdate',
+      javaCode: `
+      AuthService auth = (AuthService) x.get("auth");
+      if ( ! auth.check(x, "notificationDAO.update.*") ) throw new AuthorizationException("You don't have permission to update notifications you do not own.");
+      `
+    },
+    {
+      name: 'authorizeOnDelete',
+      javaCode: `
+      AuthService auth = (AuthService) x.get("auth");
+      if ( ! auth.check(x, "*") ) throw new AuthorizationException("You don't have permission to delete notifications you do not own.");
+      `
+    },
+    {
+      name: 'authorizeOnRead',
+      javaCode: `
+      AuthService auth = (AuthService) x.get("auth");
+      if ( ! checkOwnership(x) && ! auth.check(x, "notificationDAO.read.*") ) throw new AuthorizationException("You don't have permission to read notifications you do not own.");
```

Comment:
you don't need to check for global read permission here, `AuthorizableAuthorizer.checkGlobal` will do it for you. This method won't be even called if user has `notification.read.*` in these methods you actually need to check for `read.id`

Reply:
oh true true